atomic_reactor/plugins/post_compare_components.py

```diff
@@ -0,0 +1,106 @@
+"""
+Copyright (c) 2017 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+"""
+from atomic_reactor.plugin import PostBuildPlugin
+from atomic_reactor.constants import (PLUGIN_COMPARE_COMPONENTS_KEY,
+                                      PLUGIN_FETCH_WORKER_METADATA_KEY)
+
+
+SUPPORTED_TYPES = ("rpm",)
+
+
+class CompareComponentsPlugin(PostBuildPlugin):
+    """
+    Compare components from each worker build and verify the same version was
+    on each worker.
+    """
+
+    key = PLUGIN_COMPARE_COMPONENTS_KEY
+    is_allowed_to_fail = False
+
+    def rpm_compare(self, a, b):
+        """
+        Compare rpm component version 'a' with 'b'.
+        'name' is implied equal as it is the key() lookup
+        """
+        if a['version'] == b['version'] and \
+           a['release'] == b['release'] and \
+           a['signature'] == b['signature']:
+            return
+
+        raise ValueError("%s != %s" % (a, b))
+
+    def get_component_list(self, worker_metadatas):
+        """
+        Find the worker build's component list.
+        The strategy is to find the 'output' key in the metadata and then
+        search the array for a 'docker-image' type.  The 'log' types are not
+        interesting.
+
+        :return: array of components
```

Comment:
I think this docstring could do with tightening up.

This is returning a list of lists of "component" dicts. The method name "get_component_list" suggests returning a list of "components". Should it be "get_components_from_workers" instead?

The doc string says "Find the worker build's component list", making it seem like there is a single worker build. Should it say "Find the components from each worker build" instead?

The ":return:" is inaccurate. It's a list of component lists.